### PR TITLE
feat: add cross process GPU locking

### PR DESCRIPTION
**Pull request by vmx**
This PR extracts the GPU locking out of https://github.com/filecoin-project/bellman/pull/30. It's the same code with the only functional change that it propagates up a result instead of panicking.

### CODE REVIEW

File: src/gpu/mod.rs [hunk 1]
```diff
@@ -37,3 +37,26 @@ use ocl::Device;
...
+pub struct LockedFile(File);
+
+use std::fs::File;
+
+use std::io;
+pub const LOCK_NAME: &str = "/tmp/bellman.lock";
+pub fn lock() -> io::Result<LockedFile> {
+    let file = File::create(LOCK_NAME)?;
+
+    #[cfg(feature = "gpu")]
```
Comment (dignifiedquire) [inline src/gpu/mod.rs]: this shouldn't do anything if we don't use the gpu, so that file creation should be part of the cfg block

Comment (vmx) [reply src/gpu/mod.rs]: I've pushed a new commit with a hopefully cleaner approach.

File: src/gpu/utils.rs [hunk 2]
```diff
@@ -62,3 +64,20 @@ pub fn get_core_count(d: Device) -> GPUResult<usize> {
...
     }
 }
+
+#[derive(Debug)]
+pub struct LockedFile(File);
+
+pub const LOCK_NAME: &str = "/tmp/bellman.lock";
+
+pub fn lock() -> io::Result<LockedFile> {
+    let file = File::create(LOCK_NAME)?;
```
Comment (dignifiedquire) [inline src/gpu/utils.rs]: One more request, lets write an `info` log when locking and unlocking, just so we have a chance to debug this if it doesn't work as expected.

**Comment by vmx** [comment]
I had to sneak in one more change. I renamed `lck` to `lock` (soooo much better).

**Pull request merged**